vivarium/core/serialize.py

```diff
@@ -340,103 +153,99 @@ def deserialize(  # pylint: disable=arguments-differ
         return unit_data
 
 
-class ProcessSerializer(Serializer):
-    """Serializer for processes.
-
-    Currently only supports serialization (for emitting simulation
-    configs).
+class NumpySerializer(Serializer):
+    """Serializer for Numpy arrays.
+    Numpy array serialization is lossy--we serialize to Python lists, so
+    deserialization will produce a Python list instead of a Numpy array.
     """
 
-    def can_serialize(self, data: Any) -> bool:
-        return isinstance(data, Process)
+    def __init__(self) -> None:
+        super().__init__()
 
-    def serialize_to_string(self, data: Process) -> str:
-        """Create a dictionary of process name and parameters."""
-        return str(dict(data.parameters, _name=data.name))
+    class Codec(TypeEncoder):
+        python_type = np.ndarray
+        def transform_python(self, value: np.ndarray) -> List:
+            return value.tolist()
 
-    def can_deserialize(self, data: Any) -> bool:
-        return False
-
-    def deserialize(self, data: Any) -> None:
+    def deserialize_from_string(self, data: str) -> None:
         raise NotImplementedError(
-            'Processes cannot be deserialized.')
+            f'{self} cannot be deserialized.')
```

Comment:
This is going to print `NumpySerializer cannot be deserialized`, which is confusing because it's not a NumpySerializer being deserialized. Maybe also worth noting that this should never happen since we never serialize something in the format that would cause this deserializer to be called.

Reply:
Removed in 9860d8a.